core/src/marker/markerManager.cpp

```diff
@@ -297,6 +298,27 @@ bool MarkerManager::buildStyling(Marker& marker) {
     try {
         // Update the draw rule for the marker.
         YAML::Node node = YAML::Load(marker.stylingString());
+
+        // Check if marker styling defines a layer in the yaml scene
+        if (Node layer = node["layer"]) {
+            if (!layer.IsScalar()) {
+                LOGE("layer for marker styling needs to be a scalar value.");
+                return false;
+            }
+
+            const DataLayer* sceneLayer = m_scene->findLayer(layer.Scalar());
+            if (sceneLayer && !sceneLayer->rules().empty()) {
+                // select first drawRule from this layer's rules
+                auto rule = sceneLayer->rules().front();
```

Comment:
Using only the first draw rule for a layer is also quite limiting. If at all possible, we should use the same rule merging and application process that we use for non-marker geometry, to minimize potential confusion.